src/Models/Product.php

```diff
@@ -134,6 +136,16 @@ public function rewrites(): HasMany
             ->where('entity_type', 'product');
     }
 
+    public function parent(): HasOneThrough
+    {
+        return $this->hasOneThrough(
+            config('rapidez.models.product'),
+            config('rapidez.models.product_link'),
+            'product_id', 'entity_id',
+            'entity_id', 'parent_id'
+        )->withoutGlobalScopes();
```

Comment:
Maybe if we follow some more key conventions and some relations on the link model we can remove some of those keys.

Reply:
The third key is technically unnecessary in the current state, but if we turn the ProductLink model into a pivot with `asPivot` and manually set $foreignKey to `entity_id` in the constructor we can also remove the second key, turning it into:

```
return $this->hasOneThrough(
    config('rapidez.models.product'),
    config('rapidez.models.product_link),
    firstKey: 'product_id',
    secondLocalKey: 'parent_id',
)->withoutGlobalScopes();
```

I've tried some other things but can't really find anything else possibly worthwhile. The Laravel documentation on this stuff seems honestly kind of lacking and hard to parse.